### DISCUSSION TITLE
[Issue]: Needed advice from pro autogen users on app architecture

### DISCUSSION THREAD

**Discussion by NikolayTV** [microsoft/autogen#2460] [Q&A]
### Describe the issue

Hi! I am working on an app with quite long conversation scenario. I have a working PoC with custom code, but its a bit of a mess, and now trying to re-implement it using autogen. I attached an architecture of what I am trying to build. 
* Is it right place here to ask this question btw? Sorry for so many words.

Very much needed advice on how to assemble it with this lib! Would appreciate any reference on app example or advice. 

I was looking into examples for a some days, read most of the docs for more than 1 time, but still a bit confused on variety of possibilities.

### Here is my App flow
1) Coordinator chooses specialist (Nurse or Doctor) and instructs them on what to say to the Human based on <CONVERSATION_SCENARIO> and <HealthCard_json> in system message. 
2) Nurse asking symptoms and Doctor provides consultation to the Human
3) Human responds, and HealthCard_parser runs in backround to update HealthCard from human answers, which is saved to database
4) Coordinator`s system message (on the first position in messages) gets updated with new fields in healthCard, and based on updated <HealthCard_json> and <CONVERSATION_SCENARIO>  he decides on what to do next

* parser needs to see only 2 last dialogs between specialist and human (for token economy)
* first system message of coordinator needs to be updated after every cicle

### My questions:
1) I am confused on what conversational pattern should I use here - GroupChat, or user.initiate_chat with list of agents defined as dicts, or to combine it both in some way. 
2) How to integrate HealthCard as a "living" variable that gets updated after parser answer with function and inserted into system message?
3) How do I have precise control with code over what goes from parser to coordinator? (I want to use python function to modify user answer with correctly loaded json)
4) What do you think about updating 1st system message in a messages list? (Will it be difficult technically? assuming the idea is working, I suppose) 

<img width="882" alt="image" src="https://github.com/microsoft/autogen/assets/32233366/13e68444-c08b-48bf-b354-3fee2f2d58a1">

Is it right place here to ask this question btw? Sorry for so many words.

**Comment by ekzhu** [comment]
This is a good question. This coordinator design reminds me of the "active group chat manager" concept @gagb @afourney 

> Is it right place here to ask this question btw? Sorry for so many words.

I transferred this to discussion. In the future question like this is good for discussions.
